companion-api/api/user.go

```diff
@@ -1,54 +1,208 @@
 package api
 
 import (
-	"crypto/sha256"
-	"encoding/hex"
 	"net/http"
 
 	restful "github.com/emicklei/go-restful"
 	"github.com/mcluseau/autorizo/api"
+	"github.com/mcluseau/autorizo/auth"
+	"github.com/mcluseau/autorizo/companion-api/backend"
 )
 
 // Register provide a restful.WebService from this API
 func (cApi *CompanionAPI) registerUsers(ws *restful.WebService) {
 	ws.
-		Route(ws.POST("/user").
-			To(cApi.createOrUpdateUser).
-			Doc("Create or update user.").
+		Route(ws.POST("/user/{user-id}").
```

Comment:
Use plural ;) `POST("/users")` and don't have the ID in the URL for this usage. It should read this kind of struct:
```golang
type CreateUserReq struct {
  ID string
  UserData
}
```